lib/Runtime/Library/JavascriptProxy.cpp

```diff
@@ -919,11 +912,38 @@ namespace Js
         Var trapResult = getEnumeratorMethod->CallFunction(arguments);
         threadContext->SetImplicitCallFlags((Js::ImplicitCallFlags)(saveImplicitCallFlags | ImplicitCall_Accessor));
 
-        if (!JavascriptOperators::IsObject(trapResult))
+        if (!JavascriptOperators::IsArray(trapResult))
         {
-            JavascriptError::ThrowTypeError(scriptContext, JSERR_InconsistentTrapResult, _u("enumerate"));
+            JavascriptError::ThrowTypeError(scriptContext, JSERR_InconsistentTrapResult, _u("ownKeys"));
         }
-        *enumerator = IteratorObjectEnumerator::Create(scriptContext, trapResult);
+
+        JavascriptArray* arrTrapResult = JavascriptArray::FromVar(trapResult);        
+        uint32 len = arrTrapResult->GetLength();
```

Comment:
checked above: JavascriptOperators::IsArray
do you mean we need to check with JavascriptArray::Is()?